llvm/include/llvm/IR/Intrinsics.td

```diff
@@ -1370,7 +1370,7 @@ let IntrProperties = [IntrNoMem, IntrSpeculatable, IntrWillReturn] in {
 
 // The result of eh.typeid.for depends on the enclosing function, but inside a
 // given function it is 'const' and may be CSE'd etc.
-def int_eh_typeid_for : Intrinsic<[llvm_i32_ty], [llvm_ptr_ty], [IntrNoMem]>;
+def int_eh_typeid_for : Intrinsic<[llvm_i32_ty], [llvm_anyptr_ty], [IntrNoMem]>;
 
 def int_eh_return_i32 : Intrinsic<[], [llvm_i32_ty, llvm_ptr_ty]>;
```

Comment:
Doesn't this intrinsic need the same llvm_anyptr_ty overload?

Reply:
Possibly, but it's a slightly different matter which I should probably deal with in another PR, wherein we handle the fact that the Program AS needn't be 0. AFAICT these intrinsics are here to support GCC's `__builtin_eh_return`, and the pointer arg is supposed to point to a handler function in the Program AS. I think it's preferable to handle that topic independently since it'll spawn its own lively discussion, no doubt.